src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcStatementHandle.cs

```diff
@@ -39,22 +35,17 @@ internal SQLLEN(IntPtr value)
             return new SQLLEN(value);
         }
 
-        public static unsafe implicit operator int(SQLLEN value)
+        public static implicit operator int(SQLLEN value)
         {
-#if WIN32
-            return (int)value._value.ToInt32();
-#else
-            long l = (long)value._value.ToInt64();
-            return checked((int)l);
-#endif
+            return value._value.ToInt32();
```

Comment:
IntPtr.ToInt32() will throw if the stored value is out of range of an int32. (In 32-bit proc, this check is skipped because it always succeeds.)